### DISCUSSION TITLE
Correctness of chapter on callPackage

### DISCUSSION THREAD

**Discussion by boblehest** [ryan4yin/nixos-and-flakes-book#36] [General]
Hi, thanks for the fantastic resource on all things Nix! I'm reading the chapter on callPackage and I'm wondering if some details are incorrect or misleading:

In the [second paragraph](https://github.com/ryan4yin/nixos-and-flakes-book/blob/0b6065ed621b265f61d2703f8a7385760ca4602b/docs/nixpkgs/callpackage.md?plain=1#L5) it states that the imported file "must be a Derivation or a function that returns a Derivation.", but I don't think the former case is valid (as far as I see, the imported file must be a function). Less importantly, it also states that you should give `callPackage` a file as input, but this is not the case; directly giving a function is fine too.

The [fourth paragraph](https://github.com/ryan4yin/nixos-and-flakes-book/blob/0b6065ed621b265f61d2703f8a7385760ca4602b/docs/nixpkgs/callpackage.md?plain=1#L9) starts to explain what happens _if_ it the imported file gives a function, leaving me to expect an explanation of the other case to come later, but it never does explain any other cases.

**Comment by ryan4yin** [comment]
Indeed, I did misunderstand here, I thought `callPakcage` was similar to `import` before, but it seems that's not actually the case.

**Comment by ryan4yin** [comment]
Thanks for the feedback ❤️ I have updated the content. Please take a look when you are free.

**Reply by boblehest** [reply]
I really like the new text, thank you. :)
